### PR TITLE
bindinfo: avoid duplicate bindings caused by concurrent baseline capture (#22182)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #22182 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr 22295
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/22295:ti-srebot:release-4.0-0a4ec0da2b80
```

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/22178

Problem Summary:

Duplicate bindings for a same original_sql.

### What is changed and how it works?

What's Changed:

Call `CreateBindRecord` instead of `AddBindRecord` in `CaptureBaselines`.

How it Works:

`AddBindRecord` only checks the duplicate in binding cache, while `CreateBindRecord` would remove duplicate bindings in `mysql.bind_info` unconditionally.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects
N/A

### Release note 

- Avoid duplicate bindings caused by concurrent baseline capture

### CODE REVIEW

File: bindinfo/bind_test.go [hunk 1]
```diff
@@ -1761,6 +1761,54 @@ func (s *testSuite) TestDMLIndexHintBind(c *C) {
 	c.Assert(tk.MustUseIndex("delete from t where b = 1 and c > 1", "idx_c(c)"), IsTrue)
 }
 
+<<<<<<< HEAD
```
Comment (qw4990) [inline bindinfo/bind_test.go:1764]: PTAL @eurekaka

**Review by Reminiscent — COMMENTED** [review]
LGTM

**Review by qw4990 — APPROVED** [review]
LGTM

**Pull request merged**